design/airgap-existing-cluster.md

```diff
@@ -0,0 +1,109 @@
+# Support for installing KOTS Admin Console in Airgapped Clusters
+
+## Goals
+
+- Install kotsadm to an existing cluster where the cluster does not have internet access
+- Install kotsadm to an existing cluster from a workstation where the workstation does not have internet access at the time of installation
+- Support different credentials for push / pull (workstation / cluster) to the local registry
+- Support temporary/dynamic and persistent/static credentials for pushing kotsadm images to the registry
+- Support only persistent/static credentials for pulling kotsadm images to the registry
+- Support different credentials for pushing vs pulling images to the local registry
+
+## Non Goals
+
+- Create a single installer for kotsadm + application
+- Create a single download for kotsadm, kots plugin, application and license
+- Automated installs of kotsadm to existing clusters in airgap mode
+- Support for installing versions of KOTS different from the KOTS CLI version installed
+
+## Background
+
+It's not currently possible to install KOTS Admin Console into an existing cluster that doesn't have outbound internet access.
+
+## High-Level Design
+
+Airgap Workstation: the computer with kubectl access to the airgap cluster
+Public Workstation: the computer with access to the internet
+
+From the public workstation:
+1. The user downloads the kots plugin suitable for Airgap Workstation.
+1. The user downloads the kotsadm archive containing kots container images
```

Comment:
I'm not clear on this.  Does the user download the archive from github or does the "kots install" command download everything for the user?

Reply:
We were vague there. :) Worst case, the vendor would send an email to the admin with a link to download. They might add a redirector or cname but it's a straight download. It is **not** downloaded via `kots install`.

It's conceivable to combine downloading the kots plugin and kotsadm archive into the same download tarball, and that would probably be a better experience, but it's not a requirement. Heck, it would be better to download all three things (those two plus the application airgapped bundle) in a single download. But we're trying to limit scope here to a minimal iteration with a passable workflow. We could probably enhance that later.